packages/server/src/orchestrator.ts

```diff
@@ -119,7 +118,9 @@ export function* createOrchestrator(options: OrchestratorOptions): Operation {
       console.debug('[orchestrator] connection server ready');
     });
     yield fork(function*() {
-      yield appServerDelegate.receive({ status: 'ready' });
+      yield options.atom.once((state) => {
+        return state.appService.appStatus === 'started'
```

Comment:
We don't want to continue here until the app is reachable. I think that we probably ought to have a separate status for the app and the app service no? The app service is pretty much running the whole time, even through multiple app restarts

Reply:
To confirm, we don't want the orchestrator to be ready until the app is reachable? This introduces different behaviour on start than once the orchestrator's already running:
- on start an unreachable URL will cause the orchestrator to never reach a ready state
- once the orchestrator is already running / ready it's possible to update the app url to an unreachable one, while the orchestrator continues be in a running / ready state

On separating state - who would be consuming the app service state? I guess I'm having a hard time formulating the app service states without knowing its purpose.

Reply:
Hmm..... these are really interesting questions. In my mind, the idea of a service being ready is simply that it is available to accept requests. In the case of the app service, it is available to service requests to set the application options, so this actually seems right the way you have it. The problem is that the `bigtest ci` command, and the test cases assume that when the orchestrator sends the ready event, that it is ready to run tests.

Of course, because we're using stateful mailboxes, the request to run will wait until the orchestrator does become ready so this should not technically be a problem, but does pose a conundrum around what exactly is "ready"

On the subject of services themselves, I'm looking ahead towards them to have a uniform shape across all of the services (app server, browser manager, manifest-generator, etc...). That way, we can build UI to inspect services in a generic way, so that if some service has an error it can be surfaced without anything specific to the service itself. Basically, I'm imagining what it would look like to build an inspector for the entire orchestrator state.